onnx/defs/rnn/defs.cc

```diff
@@ -56,16 +59,31 @@ via some custom implementation such as CuDNN.
 `RBi` - R recurrence weight matrix for backward input gate
 `WBbi` - WR bias vectors for backward input gate
 `RBbi` - RR bias vectors for backward input gate
-`ReLU(X)` - max(X, 0)
-`tanh(X)` - hyperbolic tangent of X
 `H` - Hidden state
 `num_directions` - 2 if direction == bidirectional else 1
 
+Activation functions:
+  ReLU(x)                - max(0, x)
+  tanh(x)                - (1 - e^{-2x})/(1 + e^{-2x})
+  sigmoid(x)             - 1/(1 + e^{-x})
+  (Below are optional)
+  linear(x)              - alpha*x + beta
+  leakyReLU(x)           - x if x >= 0 else alpha * x
+  thresholdedReLU(x)     - x if x >= alpha else 0
+  PReLU(xi)              - xi if xi >= 0 else alpha[i]* xi over dim 0
+  scaledTanh(x)          - alpha*tanh(beta*x)
+  sigmoidHard(x)         - min(max(alpha*x + beta, 0), 1)
+  ELU(x)                 - x if x >= 0 else alpha*(e^x - 1)
+  softsign(x)            - x/(1 + |x|)
+  softplus(x)            - log(1 + e^x)
+  parametricSoftplus(xi) - alpha[i]*log(1 + e^{beta[i]* xi}) over dim 0
+
 Equations:
-  - Ht = Activation(Wi*Xt + Ri*Ht-1 + Wbi + Rbi)
+  - Ht = Activation((Xt^T)*Wi + Ht-1*Ri + Wbi + Rbi)
 )DOC")
-    .Attr("activation", "One (or two if bidirectional) activation function for "
-          "input gate. It must be one of tanh and ReLU. Default `tanh`.",
+    .Attr("activations", "One (or two if bidirectional) activation function for "
+          "input gate. The activation function must be one of the activation "
+          "functions specified above. Default `tanh`.",
           AttrType::STRING)
     .Input(1, "W",
```

Comment:
Do we actually expect this to be [1, hidden_size, input_size] for forward only pass ? If yes, this will require expand_dimensions / squeeze operations. If no, and we actually expect [hidden_size, input_size] for forward and [2, hidden_size, input_size] for bidirectional, then the spec should reflect this.

Reply:
Another option is to make it [num_directions*hidden_size, input_size]

Reply:
Let us leave it as is for now.